conans/test/utils/scm.py

```diff
@@ -48,6 +59,17 @@ def create_local_git_repo(files=None, branch=None, submodules=None, folder=None,
     return tmp.replace("\\", "/"), git.get_revision()
 
 
+def git_change_and_commit(files, folder, msg="fix"):
+    save_files(folder, files)
+    cwd = os.getcwd()
+    try:
+        os.chdir(folder)
```

Comment:
I think `chdir` might be avoided somehow, e.g. using `git -C <folder>` within commands
it should simplify the code, and should left cwd untouched in case conan is crashed/aborted